app/queries/get_activity_log_events.rb

```diff
@@ -15,25 +15,44 @@ class GetActivityLogEvents
     ],
   }.freeze
 
-  CHANGES_TO = %w[
+  INCLUDE_APPLICATION_CHOICE_CHANGES_TO = %w[
     reject_by_default_feedback_sent_at
     offer_changed_at
   ].freeze
 
+  IGNORE_STATUS = %i[interviewing].freeze
+
   def self.call(application_choices:, since: nil)
     since ||= Time.zone.local(2018, 1, 1) # before the pilot began, i.e. all records
 
-    filter = CHANGES_TO.map { |f| "jsonb_exists(audited_changes, '#{f}')" }
-                      .join(' OR ')
+    application_choice_audits_filter = INCLUDE_APPLICATION_CHOICE_CHANGES_TO.map { |f|
+      "jsonb_exists(audited_changes, '#{f}')"
+    }.join(' OR ')
+
+    filtered_statuses = ApplicationStateChange::STATES_VISIBLE_TO_PROVIDER - IGNORE_STATUS
+    status_transitions_to_include = filtered_statuses.map { |status| "'#{status}'" }.join(', ')
+
+    application_choice_audits_filter += " OR audited_changes::json#>>'{status, 1}' IN (#{status_transitions_to_include})"
+
+    Audited::Audit.includes(INCLUDES).from <<~COMBINE_AUDITS_WITH_APPLICATION_CHOICES_SCOPE_AND_FILTER.squish
+      (
+        SELECT a.*
+          FROM audits a
 
-    statuses_visible_to_providers = ApplicationStateChange::STATES_VISIBLE_TO_PROVIDER.map { |status| "'#{status}'" }.join(', ')
-    filter += " OR audited_changes::json#>>'{status, 1}' IN (#{statuses_visible_to_providers})"
+          INNER JOIN (#{application_choices.to_sql}) ac
+            ON (
+              auditable_type = 'ApplicationChoice'
+              AND auditable_id = ac.id
+              AND action = 'update'
+              AND ( #{application_choice_audits_filter} )
+            ) OR (
+              associated_type = 'ApplicationChoice'
+              AND associated_id = ac.id
```

Comment:
I noticed previously we were filtering by `action` being `create` or `update`, is it ok to include all actions for associated type?

Reply:
I think this omission is OK here, at least for interviews as it's a soft-deleted model, all events should implicitly be `create` or `update`.

Reply:
Yes, that's intentional. We exclude `create` events when the event directly related to the ApplicationChoice because this corresponds to the inital model creation, when the candidate adds courses to their application. At this stage application choices are at the `:unsubmitted` stage and should not be visible to providers. The first event a provider user should see is application submission, which is an `update`.

For interviews, on the other hand, and any such associated model, a `create` corresponds to the initial scheduling of an interview, which should be visible.